charts/spire/charts/spire-agent/templates/daemonset.yaml

```diff
@@ -74,6 +74,36 @@ spec:
         {{- toYaml .Values.hostAliases | nindent 8 }}
       {{- end }}
       initContainers:
+        {{- if eq .Values.workloadAttestors.k8s.verification.type "hostCert" }}
+        - name: gather-host-cert
+          securityContext:
+            {{- $mainSecurityContext | toYaml | nindent 12 }}
+          #FIXME socketAlternate
+          #image: {{ template "spire-lib.image" (dict "image" .Values.socketAlternate.image "global" .Values.global) }}
+          image: chainguard/min-toolkit-debug:latest@sha256:eb1c3f94b5a2303d74a804c1bfd5adc9c98e55e344bbe1adcd6522d6e80d5733
+          imagePullPolicy: {{ .Values.socketAlternate.image.pullPolicy }}
+          command: ["bash", "-xec"]
+          args:
+            - |
+              {{- if ne .Values.workloadAttestors.k8s.verification.hostCert.fileName "" }}
+              openssl x509 -in /host/{{ .Values.workloadAttestors.k8s.verification.hostCert.fileName | quote }} -out /hostCert/kubelet.crt
```

Comment:
Its reading through the symlink, which I think should work.

Say we take the openshift layout as an example.

/host here would be mounted from /var/lib/kubelet/pki on the host.

so, if fileName = kubelet-server-current.pem and kubelet-server-current.pem is symlinked to kubelet-server-12345.pem in the same directory, it would look at  /var/lib/kubelet/pki/kubelet-server-current.pem on the host, then redirect to /var/lib/kubelet/pki/kubelet-server-12345.pem and pull out the cert into the emptydir.

Reply:
it does. it actually tries to assume that its part of the container file system. When i changed from `/host` to `/var/lib/kubelet/pki` it worked without issue

Reply:
oh. is the symlink a full filesystem path?

Reply:
Yup

```
kubelet-server-current.pem -> /var/lib/kubelet/pki/kubelet-server-2024-08-13-12-27-02.pem
```

Reply:
Ah. I see. ok. so, we should probably just mount it in the container at the same place its pulled off the host.

Reply:
Fixed